SIGSEGV: on transfert on https

Hi,

**What happened**: coredns stops with SIGSEGV on AXFR requests.


**What you expected to happen**: get zone via.
<code>~/go/bin/q -i @https://x.x.x.x -v example.org -t axfr</code>

**How to reproduce it (as minimally and precisely as possible)**:
<pre>

https://example.org:443 {
tls public.pem privkey.pem
file db.example.org
transfer {
 to *
 }

log
errors
}
</pre>

**Anything else we need to know?**: I'm running coredns from alpine on docker.
<pre>
/coredns/coredns --conf coredns
https://example.org.:443
CoreDNS-1.10.1
linux/amd64, go1.19.9,
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9da424]

goroutine 53 [running]:
github.com/coredns/coredns/core/dnsserver.(*DoHWriter).TsigTimersOnly(0x0?, 0x0?)
	<autogenerated>:1 +0x24
github.com/miekg/dns.(*Transfer).Out(0x0?, {0x14c2be0, 0xc0002085c0}, 0xc00031f9e0, 0x0?)
	/root/go/pkg/mod/github.com/miekg/dns@v1.1.54/xfr.go:220 +0x72
github.com/coredns/coredns/plugin/transfer.(*Transfer).ServeDNS.func1()
	/coredns/plugin/transfer/transfer.go:119 +0x36
created by github.com/coredns/coredns/plugin/transfer.(*Transfer).ServeDNS
	/coredns/plugin/transfer/transfer.go:118 +0x66c
</pre>

**Environment**:

- the version of CoreDNS:  CoreDNS-1.10.1
- OS (e.g: `cat /etc/os-release`): Alpine Linux v3.17


Thanks.

--- a/core/dnsserver/https.go
+++ b/core/dnsserver/https.go
@@ -4,13 +4,11 @@ import (
 	"net"
 	"net/http"
 
-	"github.com/coredns/coredns/plugin/pkg/nonwriter"
+	"github.com/miekg/dns"
 )
 
-// DoHWriter is a nonwriter.Writer that adds more specific LocalAddr and RemoteAddr methods.
+// DoHWriter is a dns.ResponseWriter that adds more specific LocalAddr and RemoteAddr methods.
 type DoHWriter struct {
-	nonwriter.Writer
-
 	// raddr is the remote's address. This can be optionally set.
 	raddr net.Addr
 	// laddr is our address. This can be optionally set.
@@ -18,13 +16,50 @@ type DoHWriter struct {
 
 	// request is the HTTP request we're currently handling.
 	request *http.Request
+
+	// Msg is a response to be written to the client.
+	Msg *dns.Msg
+}
+
+// WriteMsg stores the message to be written to the client.
+func (d *DoHWriter) WriteMsg(m *dns.Msg) error {
+	d.Msg = m
+	return nil
+}
+
+// Write stores the message to be written to the client.
+func (d *DoHWriter) Write(b []byte) (int, error) {
+	d.Msg = new(dns.Msg)
+	return len(b), d.Msg.Unpack(b)
 }
 
 // RemoteAddr returns the remote address.
-func (d *DoHWriter) RemoteAddr() net.Addr { return d.raddr }
+func (d *DoHWriter) RemoteAddr() net.Addr {
+	return d.raddr
+}
 
 // LocalAddr returns the local address.
-func (d *DoHWriter) LocalAddr() net.Addr { return d.laddr }
+func (d *DoHWriter) LocalAddr() net.Addr {
+	return d.laddr
+}
 
-// Request returns the HTTP request
-func (d *DoHWriter) Request() *http.Request { return d.request }
+// Request returns the HTTP request.
+func (d *DoHWriter) Request() *http.Request {
+	return d.request
+}
+
+// Close no-op implementation.
+func (d *DoHWriter) Close() error {
+	return nil
+}
+
+// TsigStatus no-op implementation.
+func (d *DoHWriter) TsigStatus() error {
+	return nil
+}
+
+// TsigTimersOnly no-op implementation.
+func (d *DoHWriter) TsigTimersOnly(_ bool) {}
+
+// Hijack no-op implementation.
+func (d *DoHWriter) Hijack() {}

